consensus/src/chained_bft/block_storage/sync_manager.rs

```diff
@@ -182,6 +166,31 @@ impl<T: Payload> BlockStore<T> {
         }
         Ok(())
     }
+
+    pub async fn fast_forward_sync(
```

Comment:
Perhaps not for this PR, but rather in more general terms: would there be any need for the consensus component to remain responsive while this workflow is ongoing? It seems like this could block the event-loop for a rather long time, and this might be something you could do in parallel of it. 

Then the `(Vec<Block<T>>, Vec<QuorumCert>)` could simply be received as a message on the main select of consensus, and while the information is being fetched, it could handle other messages.

Reply:
It can't because we won't be able to do anything before a node is fully synced up.